git-setup.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+# If no project name is given
+if [ $# -eq 0 ]; then
+  # Display usage and stop
+  echo "Usage: git-setup.sh <api,console,website>"
+  exit 1
+fi
+# Loop through the requested modules
+for module in "$@"; do
+  echo "Setting up '$module' module..."
+  # Set the project git URL
+  project=$(echo "git@github.com:calcom/$module.git")
+  # Check if we have access to the module
+  if [ "$(git ls-remote "$project" 2>/dev/null)" ]; then
+    echo "You have access to '${module}'"
+    # Create the .gitmodules file if it doesn't exist
+    ([ -e ".gitmodules" ] || touch ".gitmodules") && [ ! -w ".gitmodules" ] && echo cannot write to .gitmodules && exit 1
+    # Prevents duplicate entries
+    git config -f .gitmodules --unset-all "submodule.apps/$module.branch"
+    # Add the submodule
+    git submodule add --force "git@github.com:calcom/$module.git" "apps/$module"
+    # Set the default branch to main
+    git config -f .gitmodules --add "submodule.apps/$module.branch" main
+    # Adding the subdmoule ignores the `.gitignore` so a reset is needed
+    git reset
```

Comment:
This would unstage any staged changed as well. Can we do it in a way that achieves what's needed without a sideffect

Reply:
Not that I'm aware of. This will only run once tho. Suggestions always welcome.